deploy/ControllerInit.sol

```diff
@@ -0,0 +1,171 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity >=0.8.0;
+
+import { AllocatorIlkInstance } from "lib/dss-allocator/deploy/AllocatorInstances.sol";
+
+import { IAccessControl } from "lib/openzeppelin-contracts/contracts/access/IAccessControl.sol";
+
+import { UsdsInstance } from "lib/usds/deploy/UsdsInstance.sol";
+
+import { CCTPForwarder } from "lib/xchain-helpers/src/forwarders/CCTPForwarder.sol";
+
+import { RateLimitHelpers } from "src/RateLimitHelpers.sol";
+
+import { IALMProxy }   from "src/interfaces/IALMProxy.sol";
+import { IRateLimits } from "src/interfaces/IRateLimits.sol";
+
+import { ControllerInstance } from "./ControllerInstance.sol";
+
+interface IBufferLike {
+    function approve(address, address, uint256) external;
+}
+
+interface IMainnetControllerLike is IAccessControl {
+    function FREEZER() external view returns (bytes32);
+    function LIMIT_USDC_TO_CCTP() external view returns (bytes32);
+    function LIMIT_USDC_TO_DOMAIN() external view returns (bytes32);
+    function LIMIT_USDS_MINT() external view returns (bytes32);
+    function LIMIT_USDS_TO_USDC() external view returns (bytes32);
+    function RELAYER() external view returns (bytes32);
+}
+
+interface IForeignControllerLike is IAccessControl {
+    function FREEZER() external view returns (bytes32);
+    function LIMIT_PSM_DEPOSIT() external view returns (bytes32);
+    function LIMIT_PSM_WITHDRAW() external view returns (bytes32);
+    function RELAYER() external view returns (bytes32);
+}
+
+interface IPSMLike {
+    function kiss(address) external;
+}
+
+interface IVaultLike {
+    function rely(address) external;
+}
+
+library MainnetControllerInit {
+
+    struct RateLimitData {
+        uint256 maxAmount;
+        uint256 slope;
+    }
```

Comment:
You can move this up onto the top level so it doesn't need to be redefined in each library.

Reply:
Updated